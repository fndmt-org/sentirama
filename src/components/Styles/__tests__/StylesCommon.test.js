/* eslint-env node, jest */
/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
// import 'jest-styled-components';

import GlobalStyle from '../global.styles';
import themeDefaultImport from '../basicVariables.styles';

// to test global styles in styled components nedd to create a snapshot of head html tag
// eslint-disable-next-line max-len
// https://spectrum.chat/styled-components/general/testing-createglobalstyle-with-passed-props~660bbad2-babe-4ada-bf59-23d3122f25ab

describe('GlobalStyles', () => {

    let themeDefault;

    beforeEach(() => {
        themeDefault = Object.assign({}, themeDefaultImport);
    });

    it('checks screenshot of main css resets without theme', () => {
        themeDefault = {};
        renderer.create(
            <GlobalStyle theme={themeDefault}/>,
        ).toJSON();

        expect(document.head).toMatchSnapshot();
    });

    it('checks screenshot of MobilisÂ´ default theme css resets', () => {
        renderer.create(
            <GlobalStyle theme={themeDefault} />,
        ).toJSON();

        expect(document.head).toMatchSnapshot();
    });
});
