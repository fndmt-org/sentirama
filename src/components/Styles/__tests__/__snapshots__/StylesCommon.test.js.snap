// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyles checks screenshot of MobilisÂ´ default theme css resets 1`] = `
<head>
  <style
    data-styled="active"
    data-styled-version="5.2.1"
  >
    html,body,div,span,applet,object,iframe,h1,h2,h3,h4,h5,h6,p,blockquote,pre,a,abbr,acronym,address,big,cite,code,del,dfn,em,img,ins,kbd,q,s,samp,small,strike,strong,sub,sup,tt,var,b,u,i,center,dl,dt,dd,ol,ul,li,fieldset,form,label,legend,table,caption,tbody,tfoot,thead,tr,th,td,article,aside,canvas,details,embed,figure,figcaption,footer,header,hgroup,menu,nav,output,ruby,section,summary,time,mark,audio,video,button,input{border:0;font:inherit;margin:0;padding:0;vertical-align:baseline;}
    article,aside,details,figcaption,figure,footer,header,hgroup,menu,nav,section{display:block;}
    ol,ul{list-style:none;}
    blockquote,q{quotes:none;}
    blockquote:before,blockquote:after,q:before,q:after{content:'';content:none;}
    table{border-collapse:collapse;border-spacing:0;}
    *{box-sizing:border-box;}
    html,body{height:100%;overscroll-behavior:none;}
    body{-webkit-font-smoothing:antialiased;}
    h1{margin-top:0;}
    em{font-style:italic;}
    a{-webkit-text-decoration:none;text-decoration:none;}
    svg{fill:currentColor;}
    #root{height:100%;}
    html,body,div,span,applet,object,iframe,h1,h2,h3,h4,h5,h6,p,blockquote,pre,a,abbr,acronym,address,big,cite,code,del,dfn,em,img,ins,kbd,q,s,samp,small,strike,strong,sub,sup,tt,var,b,u,i,center,dl,dt,dd,ol,ul,li,fieldset,form,label,legend,table,caption,tbody,tfoot,thead,tr,th,td,article,aside,canvas,details,embed,figure,figcaption,footer,header,hgroup,menu,nav,output,ruby,section,summary,time,mark,audio,video,button,input{border:0;font:inherit;margin:0;padding:0;vertical-align:baseline;}
    article,aside,details,figcaption,figure,footer,header,hgroup,menu,nav,section{display:block;}
    ol,ul{list-style:none;}
    blockquote,q{quotes:none;}
    blockquote:before,blockquote:after,q:before,q:after{content:'';content:none;}
    table{border-collapse:collapse;border-spacing:0;}
    *{box-sizing:border-box;line-height:1.5;}
    html,body{background-color:hsla(0,0%,100%,1);color:hsla(0,0%,10%,1);height:100%;overscroll-behavior:none;}
    html{font-family:Nunito Sans;font-size:62.5%;}
    body{-webkit-font-smoothing:antialiased;font-style:normal;}
    h1{font-size:3rem;margin-top:0;}
    h2{font-size:2.4rem;}
    h3{font-size:2rem;}
    h4{font-size:1.8rem;}
    em{font-style:italic;}
    strong{font-weight:700;}
    p{font-size:1.6rem;}
    a{color:hsla(230,95%,37%,1);-webkit-text-decoration:none;text-decoration:none;}
    svg{fill:currentColor;}
    #root{height:100%;}
  </style>
</head>
`;

exports[`GlobalStyles checks screenshot of main css resets without theme 1`] = `
<head>
  <style
    data-styled="active"
    data-styled-version="5.2.1"
  >
    html,body,div,span,applet,object,iframe,h1,h2,h3,h4,h5,h6,p,blockquote,pre,a,abbr,acronym,address,big,cite,code,del,dfn,em,img,ins,kbd,q,s,samp,small,strike,strong,sub,sup,tt,var,b,u,i,center,dl,dt,dd,ol,ul,li,fieldset,form,label,legend,table,caption,tbody,tfoot,thead,tr,th,td,article,aside,canvas,details,embed,figure,figcaption,footer,header,hgroup,menu,nav,output,ruby,section,summary,time,mark,audio,video,button,input{border:0;font:inherit;margin:0;padding:0;vertical-align:baseline;}
    article,aside,details,figcaption,figure,footer,header,hgroup,menu,nav,section{display:block;}
    ol,ul{list-style:none;}
    blockquote,q{quotes:none;}
    blockquote:before,blockquote:after,q:before,q:after{content:'';content:none;}
    table{border-collapse:collapse;border-spacing:0;}
    *{box-sizing:border-box;}
    html,body{height:100%;overscroll-behavior:none;}
    body{-webkit-font-smoothing:antialiased;}
    h1{margin-top:0;}
    em{font-style:italic;}
    a{-webkit-text-decoration:none;text-decoration:none;}
    svg{fill:currentColor;}
    #root{height:100%;}
  </style>
</head>
`;
